jax/_src/numpy/reductions.py

```diff
@@ -708,6 +708,40 @@ def _cumulative_reduction(a: ArrayLike, axis: Axis = None,
 nancumprod = _make_cumulative_reduction(np.nancumprod, lax.cumprod,
                                         fill_nan=True, fill_value=1)
 
+@implements(getattr(np, 'cumulative_sum', None))
+def cumulative_sum(
+    x: ArrayLike, /, *, axis: int | None = None,
+    dtype: DTypeLike | None = None,
+    include_initial: bool = False) -> Array:
+  check_arraylike("cumulative_sum", x)
+  x = lax_internal.asarray(x)
+  if x.ndim == 0:
+    raise ValueError(
+      "The input must be non-scalar to take a cumulative sum, however a "
+      "scalar value or scalar array was given."
+    )
+  if axis is None and x.ndim > 1:
+    raise ValueError(
+      f"The input array has rank {x.ndim}, however axis was not set to an "
+      "explicit value. The axis argument is only optional for one-dimensional "
+      "arrays.")
+  axis = axis or 0
```

Comment:
Can probably remove this
```suggestion
```

Reply:
`canonicalize_axis` doesn't handle `axis=None` though?

Reply:
Oh, I saw the `ValueError` above and thought it was already handled.

If your intent is handling `axis=None`, then you should do so explicitly rather than setting `axis = 0` for any input with `bool(axis)` evaluating to False.

Reply:
Maybe handle this as part of the `if` statement above?